app/src/main/java/com/github/lookupgroup27/lookup/model/map/Renderer.kt

```diff
@@ -17,50 +19,93 @@ class Renderer(private val context: Context) : GLSurfaceView.Renderer {
 
   private lateinit var textureManager: TextureManager
   private lateinit var skyBox: SkyBox
+  private lateinit var planet: Planet
 
   private var skyBoxTextureHandle: Int = -1 // Handle for the skybox texture
 
   /** The camera used to draw the shapes on the screen. */
   val camera = Camera()
-
+  // Temporary storage for the MVP matrix
+  private val mvpMatrix = FloatArray(16)
+  private val viewMatrix = FloatArray(16)
+  private val projectionMatrix = FloatArray(16)
+
+  /**
+   * Called when the surface is created or recreated. Initializes OpenGL settings, loads textures,
+   * and sets up the scene objects.
+   *
+   * @param unused the GL10 interface, not used
+   * @param config the EGLConfig of the created surface
+   */
   override fun onSurfaceCreated(unused: GL10, config: EGLConfig) {
-    // Set the background frame color
-    GLES20.glClearColor(0.0f, 0.0f, 0.0f, 1.0f)
-
-    GLES20.glEnable(GLES20.GL_DEPTH_TEST)
-
-    // Initialize TextureManager
-    textureManager = TextureManager(context)
-
-    // Load the skybox texture (replace with your texture resource ID)
-    skyBoxTextureHandle = textureManager.loadTexture(R.drawable.skybox_texture)
-
-    // Initialize the SkyBox
-    skyBox = SkyBox()
+    GLES20.glClearColor(0.0f, 0.0f, 0.0f, 1.0f) // Set the background color
+    GLES20.glEnable(GLES20.GL_DEPTH_TEST) // Enable depth testing
+
+    textureManager = TextureManager(context) // Initialize texture manager
+    skyBoxTextureHandle =
+        textureManager.loadTexture(R.drawable.skybox_texture) // Load skybox texture
+    skyBox = SkyBox() // Initialize the skybox
+    intializeObjects() // Initialize other renderable objects
+
+    // Initialize camera matrices
+    Matrix.setIdentityM(camera.viewMatrix, 0)
+    Matrix.setIdentityM(projectionMatrix, 0)
```

Comment:
These matrices are already initialized in the Camera class, so this can also go.